addons/analytic/models/analytic_plan.py

```diff
@@ -174,6 +174,12 @@ def _compute_children_count(self):
         for plan in self:
             plan.children_count = len(plan.children_ids)
 
+    @api.onchange('parent_id')
+    def _onchange_parent_id(self):
```

Comment:
If the only goal of the method is to raise an error, shouldn't we use `@api.constrains` instead?

Reply:
I've updated the commit message ;)

Reply:
I've read it, but what I mean is if it's really not possible to use `constrains` method for this case. I feel like `onchange` aren't really designed to check validity, but rather synchronize some other field's value with what's being changed.

Reply:
Mmmhh... What do you propose ? the `api.constrains` is not called soon enough in the stack...

Below, other examples in which raise UserError is called within an _onchange

In 17.0
```
File: /home/odoo/src/17.0/odoo/addons/stock/models/stock_move_line.py
  Functions: ['_onchange_quantity']
File: /home/odoo/src/17.0/odoo/addons/survey/wizard/survey_invite.py
  Functions: ['_onchange_emails', '_onchange_partner_ids']
File: /home/odoo/src/17.0/odoo/addons/uom/models/uom_uom.py
  Functions: ['_onchange_uom_ids']
File: /home/odoo/src/17.0/odoo/addons/point_of_sale/models/product.py
  Functions: ['_onchange_type']
File: /home/odoo/src/17.0/odoo/addons/point_of_sale/models/pos_payment_method.py
  Functions: ['_onchange_journal_id']
File: /home/odoo/src/17.0/odoo/addons/point_of_sale/models/pos_order.py
  Functions: ['_onchange_amount_all']
File: /home/odoo/src/17.0/odoo/addons/l10n_id_efaktur/models/account_move.py
  Functions: ['_onchange_l10n_id_tax_number']
File: /home/odoo/src/17.0/odoo/addons/payment/models/payment_provider.py
  Functions: ['_onchange_company_block_if_existing_transactions']
File: /home/odoo/src/17.0/odoo/addons/mrp/wizard/stock_assign_serial_numbers.py
  Functions: ['_onchange_serial_numbers']
File: /home/odoo/src/17.0/odoo/addons/mrp/models/mrp_workorder.py
  Functions: ['_onchange_date_finished']
File: /home/odoo/src/17.0/enterprise/account_asset/wizard/asset_modify.py
  Functions: ['_onchange_action']
File: /home/odoo/src/17.0/enterprise/sign/wizard/sign_send_request.py
  Functions: ['_onchange_validity']
```